lib/oli_web/components/delivery/students/students.ex

```diff
@@ -93,6 +119,38 @@ defmodule OliWeb.Components.Delivery.Students do
     end)
   end
 
+  defp maybe_filter_by_card(students, nil, _), do: students
+  defp maybe_filter_by_card(students, "", _), do: students
+
+  defp maybe_filter_by_card(students, selected_card_value, filter_by) do
+    case selected_card_value do
+      "1" ->
+        Enum.filter(students, fn student ->
+          Metrics.progress_range(student.progress) == "Low" ||
+            (is_nil(student.progress) and
+               (student.enrollment_status == filter_by and
+                  student.user_role_id == 4))
```

Comment:
Wdyt about using an existing function or defining a new one that makes the role check more readable? At first glance, it is not easy to realize what role we are using to filter here.